src/java.desktop/share/classes/sun/font/CompositeFont.java

```diff
@@ -114,8 +114,8 @@ public CompositeFont(String name, String[] compFileNames,
             components[msCnt] = fm.getEUDCFont();
             deferredInitialisation = new boolean[numSlots];
             if (defer) {
-                for (int i=0; i<numSlots-1; i++) {
```

Comment:
Do we have any jtreg tests that can reproduce the bug in the JP locale? If not, it would be good to create one based on the test described in the bug report.

Reply:
The fallback font may have legal issues and installing eudc.ttf on the test machine may be challenging.   I'll try to reproduce the issue using the EN_US locale and an open-source font.